### ISSUE TITLE
About the dwm colours

### ISSUE THREAD

**Issue by ltpour** [dylanaraps/pywal#186]
Which version of dwm are they for? I'm running version 6.1 and haven't applied any colour-related patches other than the recommended "urgent" patch. There is no `*colors` array in my config file, and instead of `norm_fg[]` I have `normfgcolor[]`, instead of `norm_bg[]` `normbgcolor[]` and so on.

**Comment by MitchWeaver** [comment]
I just cloned a fresh copy of dwm from http://git.suckless.org/dwm and it does have a *colors[] array.

Are you sure you are using the latest copy? (6.1)

**Comment by ltpour** [comment]
Yes, 6.1, but I didn't clone it with git, I installed the Gentoo package which fetches the source from [here](https://dl.suckless.org/dwm/). Looks like the 6.1 release is already 2 years old, the colors*[] array was probably added after that.

It seems the instructions and the file that pywal produces are correct for the latest git version but not for the 6.1 release (maybe it should be mentioned in the wiki?). What I did to make them work with that version was I added a bunch of #defines like `#define norm_fg normfgcolor` and `#define norm_bg normbgcolor` before the `#include` line.

**Comment by MitchWeaver** [comment]
Yes, as a dwm user myself, I do recommend staying with the most current git.

I will update the wiki now, thanks.

**Comment by ltpour** [comment]
I wouldn't exactly say the Gentoo package it "out of date" as the stable package is the latest release available and there is also a live ebuild that uses git.

**Comment by fdoerrer** [comment]
This is still an issue, for some reason the tar listed on suckless.org still doesn't feature a `*colors[][3]` array.

**Comment by SynKronus** [comment]
i do not use the urgent border patch, because I never felt much need of it, to use the colors of the pywal along with the
  dwm the only one I did was to add the required member to the enum, which already provides the necessary selection and normal colors.

below is a small patch:
`git apply wal.patch`

[wal.patch.zip](https://github.com/dylanaraps/pywal/files/2751496/wal.patch.zip)

**Comment by SynKronus** [comment]
its works :]

![2019-01-11-215515_1280x1024_scrot](https://user-images.githubusercontent.com/46579395/51065643-1f6a3f80-15ed-11e9-9e42-5a90237a11dc.png)
![2019-01-11-220922_1280x1024_scrot](https://user-images.githubusercontent.com/46579395/51065743-9e5f7800-15ed-11e9-83fb-07098a6de03e.png)
